### DISCUSSION TITLE
Confusion on what to use

### DISCUSSION THREAD

**Discussion by Notadumdum** [miguelgrinberg/python-socketio#1030] [Q&A]
Good day brother, a couple minutes ago I watched the 10th part of the socket.io tutorial which is a python Socket.io client

And i saw you used sio.call at the beginning..
![Screenshot_20220914-141003_YouTube](https://user-images.githubusercontent.com/105162586/190162950-17f2eedc-3a3b-4bfa-b592-67cf79cfa2a5.jpg)

I got a little confused so my question is
![Screenshot_20220914-135459_Kiwi Browser](https://user-images.githubusercontent.com/105162586/190163057-ddd06141-a10e-48a5-8493-3ac27433c4d9.jpg)

For a request that shows like this which method am i supposed tp use when creating socket.io client.. is it sio.emit or sio.call
Ps. I asked a question two days ago
https://github.com/miguelgrinberg/python-socketio/discussions/1021

**Comment by miguelgrinberg** [comment]
Are you reading the documentation? This is all covered there:

- [emit](https://python-socketio.readthedocs.io/en/latest/api.html#socketio.Client.emit): Emit a custom event to one **or more** connected clients.
- [call](https://python-socketio.readthedocs.io/en/latest/api.html#socketio.Client.call): Emit a custom event to a client **and wait for the response**.

I highlighted the parts the are specific to each. Which one to use depends on what you need to do.

**Reply by Notadumdum** [reply]
I am still kind of confused... because
1.the requests i want to make are the ones showing in the above picture and they are only sent once in the opening of every connection.. the only exception is ```42["tick"]```
Which gets sent from time to time.. so should i use ```call``` for ```42["tick"]``` and ```emit``` for the rest of the requests?

2.
![Screenshot_20220914-175302_Chrome](https://user-images.githubusercontent.com/105162586/190217984-1f7b9dc6-f372-4414-9c4a-39e769ce3c06.jpg)

I tried a connection like last month or thereabout https://github.com/miguelgrinberg/python-socketio/discussions/991
That is the link... in this screenshot i saw in the connect that auth is present ... so the 2nd question is should the ```42["authorization",{"session: sjsjsjjdjsjsjsjjsjsjs}]``` be put under ```sio.connect``` or it doesnt matter.. please i am a beginner

**Reply by miguelgrinberg** [reply]
The problem is that you are trying to do this looking at the low-level traffic, which you don't appear to be familiar with.

I'd say use emit for everything, because you don't have a need to get responses from any of these events.

**Reply by Notadumdum** [reply]
Okay thanks for that answer...what about the second question i asked... also how would i make the ```42["tick"]``` to be sent automatically

**Reply by miguelgrinberg** [reply]
I answered both. I said use emit for "everything". The tick event is not part of the protocol, it is application based so you have to code that yourself.

**Reply by Notadumdum** [reply]
Ok brother
Ive been disturbing a lot but this might be the last question i ask.. i hope🤞
![16631951513193803358599709519323](https://user-images.githubusercontent.com/105162586/190275354-c439b157-a844-44fe-9ab3-10da3c6e73b4.jpg)

I was looking through the main.js file and i took note of several ```keys``` like ```"emitAuth"``` in this picture
So my question is
Can i create events like this
``` py
@sio.on('emitAuth')
def on_message(data):
Print('emitAuth' , data)
sio.emit("authorization",{"session: sjsjsjjdjsjsjsjjsjsjs})
```
And also handle the rest request shown in devtools at the first picture of the main question by assigning the event name with what shows as a ```key``` in main.js or this is not really necessary

**Reply by miguelgrinberg** [reply]
I'm sorry, I don't understand what you are asking me and I'm not familiar with the partial JavaScript code you are showing me, so I don't know what it does.

**Reply by Notadumdum** [reply]
What i mean is can the ```'emitAuth'``` be used as an event name and then written like this
``` py
@sio.on('emitAuth')
def on_message(data):
Print('emitAuth' , data)
sio.emit("authorization",{"session: sjsjsjjdjsjsjsjjsjsjs})
```
The picture i sent was what showed in main.js file
...
I think the main question I am meant to be is asking is ```does the event name given really matter```

**Reply by miguelgrinberg** [reply]
Of course the event name matters. How will the other side know how to invoke the event without the name?

I hope you don't think this the wrong way, but I think you really need to invest some time and learn how Socket.IO works. Everything will become easier when you stop looking at the internals and start to think high-level.
